poetry/console/commands/add.py

```diff
@@ -86,18 +90,19 @@ def handle(self):
         if section not in poetry_content:
             poetry_content[section] = {}
 
-        for name in packages:
-            for key in poetry_content[section]:
-                if key.lower() == name.lower():
-                    pair = self._parse_requirements([name])[0]
-                    if (
-                        "git" in pair
-                        or "url" in pair
-                        or pair.get("version") == "latest"
-                    ):
-                        continue
+        existing_packages = self.get_existing_packages_from_input(
+            packages, poetry_content, section
+        )
 
-                    raise ValueError("Package {} is already present".format(name))
+        if existing_packages:
+            self.notify_about_existing_packages(existing_packages)
+
+        packages = [name for name in packages if name not in existing_packages]
```

Comment:
How would this handle scenarios like the following?

```sh
# this works today
poetry add aiohttp^2.0
poetry add aiohttp^3.0
poetry add aiohttp@latest
```

This along with cases where git/url were used the existing packages were simply ignored. The new behaviour might be regression as the requested spec is not being considered.

Reply:
As the input is simply compared to the package name in the `pyproject.toml`, this will work. Because there will be no package called `aiohttp^2.0`. There it at least for the `@latest` case a test here: https://github.com/python-poetry/poetry/blob/120cf431905c4e650d8f221d96f6ac7ffa484e52/tests/console/commands/test_add.py#L732-L734